helpers/getTransfernator.go

```diff
@@ -66,11 +67,45 @@ func createPartialGetObjects(getObject helperModels.GetObject) []ds3Models.Ds3Ge
     return partialObjects
 }
 
-func (transceiver *getTransceiver) transfer() (string, error) {
-    // create bulk get job
+func (transceiver *getTransceiver) createBulkGetJob() (*ds3Models.GetBulkJobSpectraS3Response, *[]helperModels.GetObject, error) {
+    // attempt to create a bulk get of all objects
     bulkGet := newBulkGetRequest(transceiver.BucketName, transceiver.ReadObjects, transceiver.Strategy.Options)
-
     bulkGetResponse, err := transceiver.Client.GetBulkJobSpectraS3(bulkGet)
+    if err == nil {
+        return bulkGetResponse, transceiver.ReadObjects, nil
+    }
+
+    badStatusErr, ok := err.(*ds3Models.BadStatusCodeError)
+    if !ok || badStatusErr.ActualStatusCode != http.StatusNotFound{
+        // failed to create bulk get for reason other than 404
+        return nil, nil, err
+    }
+
+    // head each item and try again for all objects that exist in the bucket
+    var objectsThatExist []helperModels.GetObject
+    for _, obj := range *transceiver.ReadObjects {
```

Comment:
Is this going to slow us down when we're heading 10k objects serially? This might hurt.  Let's say 60ms per object (swag), that gives us 10 minutes to head them before we start the download.  I'd think you'd want to do this in parallel, maybe 10-20 at a time?  that would drop the theoretical 10 minutes down to 30 seconds to 1 minute.  Presuming swag's are any good.

Reply:
We only drop down to doing the head calls if we've failed to create the bulk job with all objects due to a 404. If all the items exist, we won't slow down.

Reply:
Ahh, I missed that, thanks.